packages/grafana-data/src/utils/datasource.ts

```diff
@@ -0,0 +1,102 @@
+import { DataSourcePluginOptionsEditorProps, SelectableValue, KeyValue, DataSourceSettings } from '../types';
+
+export const onUpdateDatasourceOption = (props: DataSourcePluginOptionsEditorProps, key: keyof DataSourceSettings) => (
+  event: React.SyntheticEvent<HTMLInputElement | HTMLSelectElement>
+) => {
+  updateDatasourcePluginOption(props, key, event.currentTarget.value);
+};
+
+export const onUpdateDatasourceJsonDataOption = <J, S, K extends keyof J>(
+  props: DataSourcePluginOptionsEditorProps<J, S>,
+  key: K
+) => (event: React.SyntheticEvent<HTMLInputElement | HTMLSelectElement>) => {
+  updateDatasourcePluginJsonDataOption(props, key, event.currentTarget.value);
+};
+
+export const onUpdateDatasourceSecureJsonDataOption = <J, S extends {} = KeyValue>(
+  props: DataSourcePluginOptionsEditorProps<J, S>,
+  key: string
+) => (event: React.SyntheticEvent<HTMLInputElement | HTMLSelectElement>) => {
+  updateDatasourcePluginSecureJsonDataOption(props, key, event.currentTarget.value);
+};
+
+export const onUpdateDatasourceJsonDataOptionSelect = <J, S, K extends keyof J>(
```

Comment:
With my previous comment in mind (regarding concern about passing event as an argument) we could use the same strategy for selects, and instead of passing the `SelectableValue` just pass a value instead. Then you would not need the `*Select` variants of the `onUpdate*` functions